tests/MemoizeTest.php

```diff
@@ -16,6 +16,16 @@ public function it_should_call_return_the_value_of_the_called_function_caching_t
         $this->assertSame($previousValue, memoize($this->functionWithRandReturn(), 1, 1000000));
     }
 
+    /** @test */
+    public function it_should_reset_the_cache_if_function_to_be_executed_is_null()
+    {
+        $previousValue = memoize($this->functionWithRandReturn(), 1, 1000000);
+
+        memoize();
```

Comment:
we can do something like `memoize('reset')` and ignore the type hint for `callable`